hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -489,8 +490,12 @@ public Response get(
       responseBuilder
           .header(ACCEPT_RANGE_HEADER, RANGE_HEADER_SUPPORTED_UNIT);
 
-      if (keyDetails.getMetadata().get(ETAG) != null) {
-        responseBuilder.header(ETAG, wrapInQuotes(keyDetails.getMetadata().get(ETAG)));
+      String eTag = keyDetails.getMetadata().get(ETAG);
+      if (eTag != null) {
+        responseBuilder.header(ETAG, wrapInQuotes(eTag));
+        if (eTag.contains("-")) {
```

Comment:
To ensure that none of the conditions are missed, we can capture the `PartsCount` first (we can include this condition as part of the extract utility) and add the header to the response only when the returned string is non-null.